File: src/test/java/ru/romanow/jpa/web/utils/EntityBuilder.java
package ru.romanow.jpa.web.utils;

import org.jetbrains.annotations.NotNull;
import ru.romanow.jpa.domain.Address;
import ru.romanow.jpa.domain.Authority;
import ru.romanow.jpa.domain.Person;
import ru.romanow.jpa.domain.Role;
import ru.romanow.jpa.model.AddressInfo;
import ru.romanow.jpa.model.AuthorityInfo;
import ru.romanow.jpa.model.PersonModifyRequest;

import static java.util.stream.Collectors.toSet;
import static java.util.stream.IntStream.range;
import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
import static org.apache.commons.lang3.RandomUtils.nextInt;

public final class EntityBuilder {
    public static final int ROLE_COUNT = 2;
    public static final int AUTHORITY_COUNT = 3;

    @NotNull
    public static PersonModifyRequest buildPersonModifyRequest(int roleCount, int authorityCount) {
        return new PersonModifyRequest()
                .setFirstName(randomAlphabetic(8))
                .setMiddleName(randomAlphabetic(8))
                .setLastName(randomAlphabetic(8))
                .setAge(nextInt(18, 65))
                .setAddress(buildAddressInfo())
                .setRoles(range(0, roleCount)
                        .mapToObj(i -> randomAlphabetic(4))
                        .collect(toSet()))
                .setAuthorities(range(0, authorityCount)
                        .mapToObj(i -> new AuthorityInfo().setName(randomAlphabetic(4)).setPriority(nextInt(1, 10)))
                        .collect(toSet()));
    }

    @NotNull
    public static AddressInfo buildAddressInfo() {
        return new AddressInfo()
                .setCountry("Russia")
                .setCity("Moscow")
                .setStreet(randomAlphabetic(8))
                .setAddress(randomAlphabetic(8));
    }

    @NotNull
    public static Person buildPerson(int roleCount, int authorityCount) {
        return new Person()
                .setFirstName(randomAlphabetic(8))
                .setMiddleName(randomAlphabetic(8))
                .setLastName(randomAlphabetic(8))
                .setAge(nextInt(18, 65))
                .setAddress(buildAddress())
                .setRoles(range(0, roleCount)
                        .mapToObj(i -> new Role().setName(randomAlphabetic(4).toUpperCase()))
                        .collect(toSet()))
                .setAuthorities(range(0, authorityCount)
                        .mapToObj(i -> new Authority().setName(randomAlphabetic(4)).setPriority(nextInt(1, 10)))
                        .collect(toSet()));
    }

    @NotNull
    public static Address buildAddress() {
        return new Address()
                .setCountry("Russia")
                .setCity("Moscow")
                .setStreet(randomAlphabetic(8))
                .setAddress(randomAlphabetic(8));
    }
}

File: src/test/java/ru/romanow/jpa/utils/EntityBuilder.kt
package ru.romanow.jpa.utils

import ru.romanow.jpa.domain.Address
import ru.romanow.jpa.domain.Authority
import ru.romanow.jpa.domain.Person
import ru.romanow.jpa.domain.Role
import ru.romanow.jpa.model.AddressInfo
import ru.romanow.jpa.model.AuthorityInfo
import ru.romanow.jpa.model.PersonModifyRequest
import org.apache.commons.lang3.RandomStringUtils.insecure as string
import org.apache.commons.lang3.RandomUtils.insecure as random

const val ROLE_COUNT: Int = 2
const val AUTHORITY_COUNT: Int = 3

fun buildPersonModifyRequest(roleCount: Int, authorityCount: Int) =
    PersonModifyRequest(
        firstName = string().nextAlphabetic(8),
        middleName = string().nextAlphabetic(8),
        lastName = string().nextAlphabetic(8),
        age = random().randomInt(18, 65),
        address = buildAddressInfo(),
        roles = MutableList(roleCount) { string().nextAlphabetic(4).uppercase() }.toSet(),
        authorities = MutableList(authorityCount) {
            AuthorityInfo(name = string().nextAlphabetic(4), priority = random().randomInt(1, 10))
        }.toSet()
    )

fun buildPerson(roleCount: Int, authorityCount: Int) =
    Person(
        firstName = string().nextAlphabetic(8),
        middleName = string().nextAlphabetic(8),
        lastName = string().nextAlphabetic(8),
        age = random().randomInt(18, 65),
        address = buildAddress(),
        roles = MutableList(roleCount) { Role(name = string().nextAlphabetic(4).uppercase()) }.toMutableSet(),
        authorities = MutableList(authorityCount) {
            Authority(name = string().nextAlphabetic(4), priority = random().randomInt(1, 10))
        }.toMutableSet()
    )

fun buildAddressInfo() = AddressInfo(
    country = "Russia",
    city = "Moscow",
    street = string().nextAlphabetic(8),
    address = string().nextAlphabetic(8)
)

fun buildAddress() = Address(
    country = "Russia",
    city = "Moscow",
    street = string().nextAlphabetic(8),
    address = string().nextAlphabetic(8)
)
